pandas/tests/base/test_ops.py

```diff
@@ -137,227 +137,238 @@ def setup_method(self, method):
+
+        expected_s = Series(
+            range(len(orig), 0, -1), index=expected_index, dtype="int64"
+        )
 
-            result = o.unique()
-            if isinstance(o, Index):
-                assert isinstance(result, type(o))
-                tm.assert_index_equal(result, orig)
-                assert result.dtype == orig.dtype
-            elif is_datetime64tz_dtype(o):
-                # datetimetz Series returns array of Timestamp
-                assert result[0] == orig[0]
-                for r in result:
-                    assert isinstance(r, Timestamp)
-
-                tm.assert_numpy_array_equal(
-                    result.astype(object), orig._values.astype(object)
-                )
-            else:
-                tm.assert_numpy_array_equal(result, orig.values)
-                assert result.dtype == orig.dtype
+        result = o.value_counts()
+        tm.assert_series_equal(result, expected_s)
+        assert result.index.name is None
+
+        result = o.unique()
+        if isinstance(o, Index):
+            assert isinstance(result, type(o))
+            tm.assert_index_equal(result, orig)
+            assert result.dtype == orig.dtype
+        elif is_datetime64tz_dtype(o):
+            # datetimetz Series returns array of Timestamp
+            assert result[0] == orig[0]
+            for r in result:
+                assert isinstance(r, Timestamp)
+
+            tm.assert_numpy_array_equal(
+                result.astype(object), orig._values.astype(object)
+            )
+        else:
+            tm.assert_numpy_array_equal(result, orig.values)
+            assert result.dtype == orig.dtype
 
-            assert o.nunique() == len(np.unique(o.values))
+        # dropna=True would break for MultiIndex
+        assert o.nunique(dropna=False) == len(np.unique(o.values))
 
     @pytest.mark.parametrize("null_obj", [np.nan, None])
-    def test_value_counts_unique_nunique_null(self, null_obj):
-
-        for orig in self.objs:
-            o = orig.copy()
-            klass = type(o)
-            values = o._ndarray_values
-
-            if not allow_na_ops(o):
-                continue
-
-            # special assign to the numpy array
-            if is_datetime64tz_dtype(o):
-                if isinstance(o, DatetimeIndex):
-                    v = o.asi8
-                    v[0:2] = iNaT
-                    values = o._shallow_copy(v)
-                else:
-                    o = o.copy()
-                    o[0:2] = pd.NaT
-                    values = o._values
-
-            elif needs_i8_conversion(o):
-                values[0:2] = iNaT
-                values = o._shallow_copy(values)
+    def test_value_counts_unique_nunique_null(self, null_obj, index_or_series_obj):
+        orig = index_or_series_obj
+        o = orig.copy()
+        klass = type(o)
+        values = o._ndarray_values
+        num_values = len(orig)
+
+        if not allow_na_ops(o):
```

Comment:
@WillAyd don't we have a way to avoid using skips *inside* the test itself and rather do this externally? I don't recall exactly

Reply:
I think you can only do this via the `pytest.mark.skip` decorator. But this approach doesn't work here, because we need a condition on the parametrized fixture value. A quick Google search also didn't bring up anything better